### PR TITLE
fix: Properly size region anchor from LINE units

### PR DESCRIPTION
**Pull request by gkatsev**
CEA708 captions may set windows in LINE units, this needs to be converted to percentages to be properly displayed. In addition, adjust by the region anchor.

### CODE REVIEW

File: lib/text/ui_text_displayer.js [hunk 1]
```diff
@@ -417,11 +417,33 @@ shaka.text.UITextDisplayer = class {
...
-    regionElement.style.width = region.width + widthUnit;
     regionElement.style.position = 'absolute';
-    regionElement.style.top = region.viewportAnchorY + viewportAnchorUnit;
-    regionElement.style.left = region.viewportAnchorX + viewportAnchorUnit;
+
+    const linesUnit = shaka.text.CueRegion.units.LINES;
+    if (region.heightUnits === linesUnit && region.widthUnits === linesUnit) {
+      // from https://dvcs.w3.org/hg/text-tracks/raw-file/default/608toVTT/608toVTT.html#caption-window-size
+      regionElement.style.height = region.height * 5.33 + '%';
+      regionElement.style.width = region.width * 1.9 + '%';
```
Comment (gkatsev) [inline lib/text/ui_text_displayer.js:426]: Currently, assuming the video's aspect ratio is 16:9. 708 captions only know about 4:3 and 16:9 video. Happy to do a check here to figure out the correct multiple here but need a pointer to how to figure it out.

Comment (avelad) [reply lib/text/ui_text_displayer.js:446]: I have added a commit to this PR to calculate the aspect ratio, you just have to use the variable `this.aspectRatio_`

**Comment by gkatsev** [comment]
Provides more improvements to https://github.com/shaka-project/shaka-player/issues/2583

**Comment by avelad** [comment]
> Provides more improvements to #2583

What would be missing to have https://github.com/shaka-project/shaka-player/issues/2583 complete?

**Comment by gkatsev** [comment]
This does implement region anchors, but the issue talks about WebVTT and IMSC/TTML, and I was mostly focused on 708. It's possible that this also solves the WebVTT portion fully, but I'm not sure about the IMSC/TTML portion.

**Review by theodab — APPROVED** [review]

**Pull request merged**